### PR TITLE
[risk=low][RW-3982] Create project.rb tool to get workspace details from project ID

### PR DESCRIPTION
**Pull request by ericsong**
This adds a new project.rb tool which can take a Firecloud Project ID or workspace namespace and output workspace details (creator + collaborator emails).

Example Input
`./project.rb fetch-workspace-details --project all-of-us-workbench-test --projectId=aou-rw-test-3f23b78e`

Example output
```
***
***
***
Workspace Name: Should be seen
Workspace Namespace: aou-rw-test-3f23b78e
Creator: ericsong3@fake-research-aou.org
Collaborators:
        ericsong3@fake-research-aou.org (OWNER)
        ericsong@fake-research-aou.org (WRITER)
***
***
***
```

### CODE REVIEW

File: api/tools/src/main/java/org/pmiops/workbench/tools/FetchWorkspaceDetails.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+import org.pmiops.workbench.firecloud.model.WorkspaceAccessEntry;
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.autoconfigure.domain.EntityScan;
+import org.springframework.boot.builder.SpringApplicationBuilder;
+import org.springframework.context.annotation.Bean;
+import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
+
+/**
+ * A tool that takes an AoU username (e.g. gjordan) and fetches the FireCloud profile associated
```
Comment (jmthibault79) [inline api/tools/src/main/java/org/pmiops/workbench/tools/FetchWorkspaceDetails.java]: bad copy/paste

File: api/build.gradle [hunk 2]
```diff
@@ -627,6 +627,16 @@ task fetchFireCloudUserProfile(type: JavaExec) {
   }
 }
 
+// See project.rb command: fetch-workspace-details
+task fetchWorkspaceDetails(type: JavaExec) {
+  classpath = sourceSets.tools.runtimeClasspath
+  main = "org.pmiops.workbench.tools.FetchWorkspaceDetails"
+  systemProperties = commandLineSpringProperties
+  if (project.hasProperty("appArgs")) {
```
Comment (jaycarlton) [inline api/build.gradle]: I'm surprised you have to do this for every task that takes arguments. Also, why can't the `Eval.me()` step handle a missing property so you don't need a conditional here?

File: api/libproject/devstart.rb [hunk 3]
```diff
@@ -1124,6 +1124,46 @@ def fetch_firecloud_user_profile(cmd_name, *args)
...
+  ensure_docker cmd_name, args
+
+  op = WbOptionsParser.new(cmd_name, args)
+  op.opts.project = TEST_PROJECT
+
+  op.add_typed_option(
+      "--projectId=[projectId]",
+      String,
+      ->(opts, v) { opts.projectId = v},
+      "The Workspace to fetch details for (e.g. 'aou-rw-231823128'")
```
Comment (jaycarlton) [inline api/libproject/devstart.rb]: I'd avoid putting real identifiers into this public repo if we can. Also, it'd be good to be explicit about whether this is a FC workspace name.

Comment (ericsong) [reply api/libproject/devstart.rb]: its fake! I made it up but I wanted it to be similar enough to what a real workspace looks like so the example has some value.

Comment (jmthibault79) [reply api/libproject/devstart.rb]: Project ID, not Workspace

File: api/tools/src/main/java/org/pmiops/workbench/tools/ServiceAccountAPIClientFactory.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    return apiClient;
+  }
+
+  public WorkspacesApi workspacesApi() throws IOException {
+    WorkspacesApi api = new WorkspacesApi();
+    api.setApiClient(newApiClient(apiUrl));
+    return api;
+  }
+
+  public BillingApi newBillingApi() throws IOException {
```
Comment (gjuggler) [inline api/tools/src/main/java/org/pmiops/workbench/tools/ServiceAccountAPIClientFactory.java]: Is there a reason for the naming inconsistency betw. this and the workspacesApi() method?

Comment (ericsong) [reply api/tools/src/main/java/org/pmiops/workbench/tools/ServiceAccountAPIClientFactory.java]: nope! fixed

File: api/libproject/devstart.rb [hunk 5]
```diff
@@ -1124,6 +1124,46 @@ def fetch_firecloud_user_profile(cmd_name, *args)
   :fn => ->(*args) {fetch_firecloud_user_profile("fetch-firecloud-user-profile", *args)}
 })
 
+def fetch_workspace_details(cmd_name, *args)
+  common = Common.new
+  ensure_docker cmd_name, args
+
+  op = WbOptionsParser.new(cmd_name, args)
+  op.opts.project = TEST_PROJECT
```
[the remark this answers is not in the document]
Comment (ericsong) [reply api/libproject/devstart.rb]: TBH, I copied over most of the code from other commands but I looked into it a bit and I believe this line sets the default project to cover the case where one isn't given.

I tried running the command without providing a project and it ran with the test environment.

I then removed the line and tried running the command but got an error saying I needed to specify a project.

[the remark this answers is not in the document]
Comment (jaycarlton) [reply api/libproject/devstart.rb]: @gjuggler I'm going to start a wishlist doc for dev tooling inconsistencies, gaps, and gotchas. Please add this to that.

[the remark this answers is not in the document]
Comment (jaycarlton) [reply api/libproject/devstart.rb]: Created http://broad.io/aou-devops-tools-input for this purpose.

**Pull request merged**